subsys/bluetooth/mesh/rpl_strg_settings.c

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ * Copyright (c) 2020 Lingao Meng
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <string.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <sys/atomic.h>
+#include <sys/util.h>
+#include <sys/byteorder.h>
+
+#include <net/buf.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/mesh.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_RPL)
+#define LOG_MODULE_NAME bt_mesh_rpl_strg
+#include "common/log.h"
+
+#include "mesh.h"
+#include "net.h"
+#include "rpl.h"
+#include "settings.h"
+
+static struct bt_mesh_rpl replay_list[CONFIG_BT_MESH_CRPL];
+
+/* Replay Protection List information for persistent storage. */
+struct rpl_val {
+	uint32_t seq:24,
+	      old_iv:1;
+};
+
+static ATOMIC_DEFINE(store, CONFIG_BT_MESH_CRPL);
+
+static inline int rpl_idx(const struct bt_mesh_rpl *rpl)
+{
+	return rpl - replay_list;
+}
+
+static void clear_rpl(struct bt_mesh_rpl *rpl)
+{
+	int err;
+	char path[18];
+
+	if (!rpl->src) {
+		return;
+	}
+
+	snprintk(path, sizeof(path), "bt/mesh/RPL/%x", rpl->src);
+	err = settings_delete(path);
+	if (err) {
+		BT_ERR("Failed to clear RPL");
+	} else {
+		BT_DBG("Cleared RPL");
+	}
+
+	(void)memset(rpl, 0, sizeof(*rpl));
+	atomic_clear_bit(store, rpl_idx(rpl));
+}
+
+static void schedule_rpl_store(struct bt_mesh_rpl *entry, bool force)
+{
+	atomic_set_bit(store, rpl_idx(entry));
+
+	if (force
+#ifdef CONFIG_BT_MESH_RPL_STORE_TIMEOUT
+	    || CONFIG_BT_MESH_RPL_STORE_TIMEOUT >= 0
+#endif
+	    ) {
+		bt_mesh_settings_store_schedule(BT_MESH_SETTINGS_RPL_PENDING);
+	}
+}
+
+static void schedule_rpl_clear(void)
```

Comment:
Really there is no reason to multiply logic entities by creating a static function that consists only of recalling another function. Please remove this.